### ISSUE TITLE
Training

### ISSUE THREAD

**Issue by amauryma55** [amauryma55/python#1]
import pandas as pd
import numpy as np
from sklearn.model_selection import train_test_split
from sklearn.preprocessing import StandardScaler
from sklearn.metrics import classification_report, confusion_matrix

from sklearn.linear_model import LogisticRegression
from sklearn.ensemble import RandomForestClassifier
from xgboost import XGBClassifier
from sklearn.svm import SVC

df = pd.read_csv("tus_datos.csv")

X = df.drop('target', axis=1)  # Reemplaza 'target' por tu variable objetivo
y = df['target']

## Convertir variables categorizadas con pd.cut a dummies
X = pd.get_dummies(X, drop_first=True)

## Separar train y test
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

## Escalar (solo para modelos sensibles a escala como SVM o Logistic)
scaler = StandardScaler()
X_train_scaled = scaler.fit_transform(X_train)
X_test_scaled = scaler.transform(X_test)

## Regresión logística
lr = LogisticRegression(max_iter=1000)
lr.fit(X_train_scaled, y_train)
y_pred_lr = lr.predict(X_test_scaled)

## Random Forest
rf = RandomForestClassifier(n_estimators=100, random_state=42)
rf.fit(X_train, y_train)
y_pred_rf = rf.predict(X_test)

## XGBoost
xgb = XGBClassifier(use_label_encoder=False, eval_metric='logloss')
xgb.fit(X_train, y_train)
y_pred_xgb = xgb.predict(X_test)

## SVM
svm = SVC()
svm.fit(X_train_scaled, y_train)
y_pred_svm = svm.predict(X_test_scaled)

modelos = {
    'Logistic Regression': y_pred_lr,
    'Random Forest': y_pred_rf,
    'XGBoost': y_pred_xgb,
    'SVM': y_pred_svm
}

for nombre, y_pred in modelos.items():
    print(f"\n--- {nombre} ---")
    print(confusion_matrix(y_test, y_pred))
    print(classification_report(y_test, `y_pred))```
